File: 2562-find-the-array-concatenation-value/2562-find-the-array-concatenation-value.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findTheArrayConcVal = function(nums) {
    
    let s = 0,e = nums.length-1;
    let ans = 0;
    
    while(s <= e){
        if(s === e){
            ans += nums[s];
            break;
        } 
        ans += Number(nums[s++].toString() + nums[e--].toString())
    }
    return ans;
};

File: 2562-find-the-array-concatenation-value/2562-find-the-array-concatenation-value.cpp
class Solution {
public:
    long long findTheArrayConcVal(vector<int>& nums) {
        
        long long res = 0;
        if(nums.size() == 1) return nums[0];
        
        int s = 0,e = nums.size()-1;
        
        while(s <= e){
            if(s == e){
                res += nums[s];
                break;
            } 
            string str = to_string(nums[s++]) + to_string(nums[e--]);
            res += stoi(str);
        }
        
        return res;
    }
};